src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp

```diff
@@ -803,20 +863,25 @@ void AnalogWrapper::run()
 
                 if (useROS != ROS_disabled && rosMsgType == "geometry_msgs/WrenchStamped")
                 {
-                    yarp::rosmsg::geometry_msgs::WrenchStamped rosData;
-                    rosData.header.seq = rosMsgCounter++;
-                    rosData.header.stamp = yarp::os::Time::now();
-                    rosData.header.frame_id = frame_id;
+                    std::vector<yarp::rosmsg::geometry_msgs::WrenchStamped> rosDataVec;
+                    rosDataVec.resize(rosPublisherWrenchPortVec.size());
+
+                    for(size_t i = 0; i < rosDataVec.size(); i++)
+                    {
+                        rosDataVec.at(i).header.seq = rosMsgCounterVec.at(i)++;
+                        rosDataVec.at(i).header.stamp = yarp::os::Time::now();
+                        rosDataVec.at(i).header.frame_id = frame_idVec.at(i);
 
-                    rosData.wrench.force.x = lastDataRead[0];
-                    rosData.wrench.force.y = lastDataRead[1];
-                    rosData.wrench.force.z = lastDataRead[2];
+                        rosDataVec.at(i).wrench.force.x = lastDataRead[6 * i + 0];
+                        rosDataVec.at(i).wrench.force.y = lastDataRead[6 * i + 1];
+                        rosDataVec.at(i).wrench.force.z = lastDataRead[6 * i + 2];
 
-                    rosData.wrench.torque.x = lastDataRead[3];
-                    rosData.wrench.torque.y = lastDataRead[4];
-                    rosData.wrench.torque.z = lastDataRead[5];
+                        rosDataVec.at(i).wrench.torque.x = lastDataRead[6 * i + 3];
+                        rosDataVec.at(i).wrench.torque.y = lastDataRead[6 * i + 4];
+                        rosDataVec.at(i).wrench.torque.z = lastDataRead[6 * i + 5];
```

Comment:
Since YARP analogsensor is not typed as a force/torque sensor, it may be possible that the data the sensor is providing are not a sequence of 3force+3torques; other data may be present altogether

Example:
3forces + 3torques + temperature + 3forces + 3torques + temperature ...
3forces + 3torques + temperature + gravity + 3forces + 3torques + temperature + gravity ...

On the yarp port side we can handle this case by modifying the mapping between values and ports:
```
            <elem name="FirstSetOfChannels">0 5 0 5</elem>
            <elem name="SecondSetOfChannels">6 11 0 5</elem>
            <elem name="ThirdSetOfChannels">12 17 0 5</elem>
            <elem name="FourthSetOfChannels">18 23 0 5</elem>
```

 but there is no such freedom on the ROS side.

I suggest to add 2 parameters in the ROS group, something like `offset` and `padding` to tell the code to skip some samples and get only the meaningful data. By default they will be zero, but can be useful in some cases.

Example:
`3forces + 3torques + temperature + 3forces + 3torques + temperature ...`  -> `offset 0, padding 1` (the temperature entry is one sample to skip)
`3forces + 3torques + temperature + gravity + 3forces + 3torques + temperature + gravity ...` -> `offset 0, padding is 4` (1 temperature + 3 for gravity)

Instead, if F/T values are not the first ones (I hope this case won't happen, but since we are here):
`temperature + 3forces + 3torques + temperature + 3forces + 3torques  ...`  -> `offset 1, padding is 0` (the F/T are the last ones)
`temperature + 3forces + 3torques + gravity + temperature + 3forces + 3torques  + gravity ...` -> `offset 1, padding is 3` (skip 1 sample, get F/T and skip remaining 3)

Code may look like:
` rosDataVec.at(i).wrench.force.x = lastDataRead[ (offset + 6 + padding) * i + 0];

What do you think?

Reply:
@barbalberto that is a very good point! 
I will do the changes now. Thanks for highlighting.